lib/rules/no-extra-parens.js

```diff
@@ -448,11 +472,22 @@ module.exports = {
               node.left.type === "UnaryExpression" && isExponentiation;
             const shouldSkipRight = NESTED_BINARY && (node.right.type === "BinaryExpression" || node.right.type === "LogicalExpression");
 
-            if (!shouldSkipLeft && hasExcessParens(node.left) && (leftPrecedence > prec || (leftPrecedence === prec && !isExponentiation))) {
-                report(node.left);
+            if (!shouldSkipLeft && hasExcessParens(node.left)) {
+                if (
+                    (leftPrecedence > prec || (leftPrecedence === prec && !isExponentiation)) ||
+                    hasDoubleExcessParens(node.left)
```

Comment:
@mdjermanovic 
Yes. that makes sense.
That would be more clear. Besides, that's reasonable because the `hasDoubleExcessParens` has the duplicate logic(checking `ruleApplies`) with `hasExcessParens`